states/new_hampshire/main.py

```diff
@@ -0,0 +1,77 @@
+import json
+import scrapy
+from scrapy.crawler import CrawlerProcess
+import csv
+import re
+from io import StringIO
+from common import get_clerk_data_default_entry
+
+# We use Scrapy for New Hampshire because it has an .aspx __viewstate 
+# which must be generated and sent along with the request
+
+class NewHampshireSpider(scrapy.Spider):
+  name = 'New Hampshire'
+  start_urls = ["https://app.sos.nh.gov/Public/Reports.aspx"]
+
+  def parse(self, response):
+    yield scrapy.FormRequest.from_response(
+      response,
+      formdata={
+          'ctl00$MainContentPlaceHolder$PPReport': 'rdoCsv'
+      },
+      method="POST",
+      callback=self.generate_csv
+    )
+
+  def generate_csv(self, response):
+    f = StringIO(response.text)
+    reader = csv.DictReader(f, delimiter=",")
+    clerk_data = []
+
+    for row in reader:
+      data = self.extract_clerk_data(row)
+      if data is not None:
+        clerk_data.append(data)
+
+    with open('../../public/new_hampshire.json', 'w') as f:
+        json.dump(clerk_data, f)
+
+    return clerk_data
+
+  def extract_clerk_data(self, row):
+    town_or_city = self.extract_city_without_ward(row)
+
+    if town_or_city is None:
+      return None
+
+    clerk_data_entry = get_clerk_data_default_entry()
```

Comment:
We don't need default values since you're just going to overwrite them.  I prefer the style 

```
clerk_data_entry = {
  'city': town_or_city,
  ...
}
```